ecomap/ecomap/Model/EcomapFetcher.m

```diff
@@ -0,0 +1,1004 @@
+    
+    
+}
+
+#pragma mark - Load all Resources
+
++(void)loadResourcesOnCompletion:(void (^)(NSArray *resources, NSError *error))completionHandler
+{
+    [self dataTaskWithRequest:[NSURLRequest requestWithURL:[EcomapURLFetcher URLforResources]]
+             sessionConfiguration:[NSURLSessionConfiguration  ephemeralSessionConfiguration]
+                completionHandler:^(NSData *JSON, NSError *error) {
+                    
+                    NSMutableArray *resources = nil;
+                    NSArray *resourcesFromJSON = nil;
+                    if(!error)
+                    {
+                        //Parse JSON
+                        resourcesFromJSON = (NSArray*)[NSJSONSerialization JSONObjectWithData:JSON options:0 error:&error];
+                        resources = [NSMutableArray array];
+                        
+                        //Fill array with ECOMAPRESOURCES
+                        for(NSDictionary *resource in resourcesFromJSON)
+                        {
+                            EcomapResources *ecoRes = [[EcomapResources alloc] initWithResource:resource];
+                            [resources addObject:ecoRes];
+                            
+                            // DDLogVerbose(@"%@",resources);
+                            
+                        }
+                    }
+                    completionHandler(resources,error);
+                    
+                }];
+    
+    
+}
+
+#pragma mark - Get Problem with ID
++ (void)loadProblemDetailsWithID:(NSUInteger)problemID OnCompletion:(void (^)(EcomapProblemDetails *problemDetails, NSError *error))completionHandler
+{
+    [self dataTaskWithRequest:[NSURLRequest requestWithURL:[EcomapURLFetcher URLforProblemWithID:problemID]]
+             sessionConfiguration:[NSURLSessionConfiguration ephemeralSessionConfiguration]
+                completionHandler:^(NSData *JSON, NSError *error) {
+                    NSDictionary *problem = nil;
+                    NSArray *photos = nil;
+                    NSArray *comments = nil;
+                    
+                    EcomapProblemDetails *problemDetails = nil;
+                    NSMutableArray *problemPhotos = nil;
+                    NSMutableArray *problemComments = nil;
+                    
+                    if (!error) {
+                        //Extract received data
+                        if (JSON != nil) {
+                            //Check if we have a problem with such problemID.
+                            //If there is no one, server give us back Dictionary with "error" key
+                            //Parse JSON
+                            NSDictionary *answerFromServer = [EcomapFetcher parseJSONtoDictionary:JSON];
+                            if (answerFromServer) {
+                                //Return error. Form error to be passed to completionHandler
+                                NSError *error = [[NSError alloc] initWithDomain:NSMachErrorDomain
```

Comment:
If there is no problem on Ecomap server with N id, but we ask for it, server gives as back JSON with dictionary {"error":"no such problem"} and status code 200. But in fact it is an error for our app, so I manualy creare an error object.